The job **`pr-check-2`** failed! Check out the build log: [(web)](https://github.com/rust-lang/rust/actions/runs/24554229028/job/71786793988) [(plain enhanced)](https://triage.rust-lang.org/gha-logs/rust-lang/rust/71786793988) [(plain)](https://github.com/rust-lang/rust/commit/766773ca91c71f54fe3b26ea4474d46151ebb1ff/checks/71786793988/logs)

<details><summary><i>Click to see the possible cause of the failure (guessed by this bot)</i></summary>

```plain
357 | pub struct RangeInclusive<Idx> {
    |            ^^^^^^^^^^^^^^ ---
help: add missing generic argument
    |
431 | let mut exhausted: legacy::RangeInclusive<Idx> = 0..=0;
    |                                          +++++

error[E0308]: mismatched types
   --> library/core/src/range.rs:438:1
    |
433 | / if exhausted.empty() {
...   |
438 | | RangeInclusive::from(exhausted) // this panics
    | | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    | | |
    | | expected `()`, found `RangeInclusive<_>`
439 | | }
    | |_- `if` expressions without `else` arms expect their inner expression to be `()`
    |
    = note: expected unit type `()`
                  found struct `std::range::RangeInclusive<_>`
help: consider using a semicolon here
    |
---

error[E0599]: no method named `start` found for struct `std::range::RangeInclusive<Idx>` in the current scope
   --> library/core/src/range.rs:420:20
    |
420 | assert_eq!((single.start(), single.end()), (0, 1));
    |                    ^^^^^-- help: remove the arguments
    |                    |
    |                    field, not a method

error[E0599]: no method named `end` found for struct `std::range::RangeInclusive<Idx>` in the current scope
   --> library/core/src/range.rs:420:36
    |
420 | assert_eq!((single.start(), single.end()), (0, 1));
    |                                    ^^^ method not found in `std::range::RangeInclusive<_>`

error[E0107]: missing generics for struct `std::ops::RangeInclusive`
   --> library/core/src/range.rs:422:20
    |
422 | let empty: legacy::RangeInclusive = 0..=0;
---
357 | pub struct RangeInclusive<Idx> {
    |            ^^^^^^^^^^^^^^ ---
help: add missing generic argument
    |
422 | let empty: legacy::RangeInclusive<Idx> = 0..=0;
    |                                  +++++

error[E0599]: no method named `start` found for struct `std::range::RangeInclusive<Idx>` in the current scope
   --> library/core/src/range.rs:424:19
    |
424 | assert_eq!((empty.start(), empty.end()), (0, 0));
    |                   ^^^^^-- help: remove the arguments
    |                   |
    |                   field, not a method

error[E0599]: no method named `end` found for struct `std::range::RangeInclusive<Idx>` in the current scope
   --> library/core/src/range.rs:424:34
    |
424 | assert_eq!((empty.start(), empty.end()), (0, 0));
    |                                  ^^^ method not found in `std::range::RangeInclusive<_>`

error: aborting due to 8 previous errors

Some errors have detailed explanations: E0107, E0599, E0658.
For more information about an error, try `rustc --explain E0107`.
```

</details>

--- a/library/core/src/range.rs
+++ b/library/core/src/range.rs
@@ -402,11 +402,46 @@ impl<T> const From<RangeInclusive<T>> for legacy::RangeInclusive<T> {
 #[stable(feature = "new_range_inclusive_api", since = "1.95.0")]
 #[rustc_const_unstable(feature = "const_convert", issue = "143773")]
 impl<T> const From<legacy::RangeInclusive<T>> for RangeInclusive<T> {
+    /// Converts from a legacy range to a non-legacy range, potentially panicking.
+    ///
+    /// # Panics
+    ///
+    /// Panics if the legacy range iterator has been exhausted.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// use core::range::legacy;
+    /// use core::range::RangeInclusive;
+    ///
+    /// let single: legacy::RangeInclusive<i32> = 0..=1;
+    /// let single = RangeInclusive::from(single);
+    /// assert_eq!((single.start, single.end), (0, 1));
+    ///
+    /// let empty: legacy::RangeInclusive<i32> = 0..=0;
+    /// let empty = RangeInclusive::from(empty);
+    /// assert_eq!((empty.start, empty.end), (0, 0));
+    /// ```
+    ///
+    /// ```should_panic
+    /// use core::range::legacy;
+    /// use core::range::RangeInclusive;
+    ///
+    /// let mut exhausted: legacy::RangeInclusive<i32> = 0..=0;
+    /// exhausted.next();
+    /// # if exhausted.empty() {
+    /// # // assert!s don't work correctly in `should_panic` doctests since you
+    /// # // can't assert the panic message. Skip the rest of the test instead,
+    /// # // so that the expected panic doesn't happen and the test fails.
+    /// assert!(exhausted.is_empty());
+    /// let _ = RangeInclusive::from(exhausted); // this panics
+    /// # }
+    /// ```
     #[inline]
     fn from(value: legacy::RangeInclusive<T>) -> Self {
         assert!(
             !value.exhausted,
-            "attempted to convert from an exhausted `legacy::RangeInclusive` (unspecified behavior)"
+            "attempted to convert from an exhausted `legacy::RangeInclusive`"
         );
 
         let (start, last) = value.into_inner();

